onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -284,8 +281,15 @@ bool Attention<T>::IsPackWeightsSuccessful(int qkv_index,
                                            const T* weights_data,
                                            size_t weight_matrix_col_size,
                                            /*out*/ PrePackedWeights* prepacked_weights) {
+  bool share_prepacked_weights = (prepacked_weights != nullptr);
   size_t packb_size = MlasGemmPackBSize(head_size, input_hidden_size);
   if (packb_size == 0) {
+    if (!share_prepacked_weights) {
+      for (int i = 0; i < qkv_index; i++) {
+        packed_weights_[i].~unique_ptr();
```

Comment:
1) Would `reset()` be better ?

2) Would it be better to wrap the clean-up logic in a file-scope function to be called if any of the three packing attempts are unsuccessful (before returning Status::OK() on line 380) ? Would that potentially save duplicating this code in other places `IsPackWeightsSuccessful()` could return false in the future ?

Reply:
1. yes, sets to nullptr
2. I tried to find if there is a case it would fail in other PrePack() too, but it makes sense to add the fn() for future.

Updated.